libraries/standard/http/src/http_client.c

```diff
@@ -16,22 +45,158 @@ HTTPStatus_t HTTPClient_AddHeader( HTTPRequestHeaders_t * pRequestHeaders,
                                         int32_t rangeStart,
                                         int32_t rangeEnd )
 {
     return HTTP_NOT_SUPPORTED;
 }
 
+/*-----------------------------------------------------------*/
+
+static HTTPStatus_t _sendHttpHeaders( const HTTPTransportInterface_t * pTransport,
+                                      const HTTPRequestHeaders_t * pRequestHeaders )
+{
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    int32_t transportStatus = 0;
+
+    /* Send the HTTP headers over the network. */
+    transportStatus = pTransport->send( pTransport->pContext,
+                                        pRequestHeaders->pBuffer,
+                                        pRequestHeaders->headersLen );
+
+    if( transportStatus < 0 )
+    {
+        IotLogErrorWithArgs( "Error in sending the HTTP headers over the transport "
+                             "interface. Transport status %d.",
+                             transportStatus );
+        returnStatus = HTTP_NETWORK_ERROR;
+    }
+    else if( transportStatus != pRequestHeaders->headersLen )
+    {
+        IotLogErrorWithArgs( "Attempted to send %d of HTTP headers, transport "
+                             " reported it sent %d.",
+                             pRequestHeaders->headersLen,
+                             transportStatus );
+        returnStatus = HTTP_NETWORK_ERROR;
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static HTTPStatus_t _sendHttpBody( const HTTPTransportInterface_t * pTransport,
+                                   const uint8_t * pRequestBodyBuf,
+                                   size_t reqBodyBufLen )
+{
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    int32_t transportStatus = 0;
+
+    /* Send the HTTP body over the network. */
+    if( pRequestBodyBuf != NULL )
+    {
+        transportStatus = pTransport->send( pTransport->pContext,
+                                            pRequestBodyBuf,
+                                            reqBodyBufLen );
+
+        if( transportStatus < 0 )
+        {
+            IotLogErrorWithArgs( "Error in sending the HTTP body over the "
+                                 "transport interface. Transport status %d.",
+                                 transportStatus );
+            returnStatus = HTTP_NETWORK_ERROR;
+        }
+        else if( transportStatus != reqBodyBufLen )
+        {
+            IotLogErrorWithArgs( "Attempted to send %d of HTTP body, transport "
+                                 "reported it sent %d.",
+                                 reqBodyBufLen,
+                                 transportStatus );
+            returnStatus = HTTP_NETWORK_ERROR;
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static HTTPStatus_t _receiveHttpResponse( const HTTPTransportInterface_t * pTransport,
```

Comment:
No corresponding documentation is added for this? If not required now, can add it in future PRs

Reply:
It is at the prototype.